src/fairchem/core/trainers/base_trainer.py

```diff
@@ -357,15 +361,122 @@ def load_datasets(self) -> None:
             )
 
     def load_task(self):
-        # Normalizer for the dataset.
-        normalizer = self.config["dataset"].get("transforms", {}).get("normalizer", {})
+        # load or fit element references for dataset
+        elementrefs = (
+            self.config["dataset"].get("transforms", {}).get("element_references", {})
+        )
+        self.elementrefs = {}
+        for target in elementrefs:
+            if target == "fit" and not elementrefs["fit"].get("fitted", False):
+                otf_elementrefs = [
+                    {target: None for target in elementrefs["fit"]["targets"]}
+                ]
+                # only carry out the fit on master and then broadcast
+                if distutils.is_master():
+                    otf_elementrefs = [
+                        fit_linear_references(
+                            targets=elementrefs["fit"]["targets"],
+                            dataset=self.train_dataset,
+                            batch_size=elementrefs["fit"].get(
+                                "batch_size", self.config["optim"]["batch_size"]
+                            ),
+                            num_batches=elementrefs["fit"].get("num_batches"),
+                            num_workers=self.config["optim"]["num_workers"],
+                            max_num_elements=elementrefs["fit"].get(
+                                "max_num_elements", 118
+                            ),
+                            driver=elementrefs["fit"].get("driver", None),
+                        )
+                    ]
+                    # save the linear references for possible subsequent use
+                    if not self.is_debug:
+                        for lr_target, references in otf_elementrefs[0].items():
```

Comment:
Is this always a list of size=1? Can we just use the element in the list?

Reply:
Yeah, this is always a size one list. The only way I found to broadcast a pickleable object is with `torch.distributed.broadcast_object_list`, so I put the single object in the list and broadcast that.